### PR TITLE
Modernise examples

### PR DESCRIPTION
**Pull request by mpalmer**
Bring examples into line with current StashJS API, add a number of new examples,
and some more docs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+
+## Usage
+
+1. Make sure you've got the necessary dependencies installed:
+
+  ```bash
+  npm install
+  ```
+
+2. Have a CipherStash profile setup:
```
Comment (auxesis) [inline README.md]: Let's link here to [the docs](https://docs.cipherstash.com/tutorials/getting-started/install-stash-cli.html), so people who hit this repo have some context about what the `stash` command is, and how to login.

**Review by auxesis — CHANGES_REQUESTED** [review]
One small change requested, but this is a huge leap forward. 

Nice work @mpalmer. 🎉

File: README.md (hunk 1 above)
Comment (mpalmer) [reply README.md]: Excellent catch.  Added a commit for that.

**Review by auxesis — APPROVED** [review]
Thanks for the changes! 

LGTM.

**Pull request merged**